macros/remove_duplicate_and_prefix_from_columns.sql

```diff
@@ -0,0 +1,24 @@
+{% macro remove_duplicate_and_prefix_from_columns(columns, prefix='', exclude=[]) %}
+        
+        {# Loop once to find duplicates #}
+        {%- set duplicate_exclude = [] -%}
+        {% for col in columns if col.name not in exclude %}
+        {%- for dupe in columns if col.name[prefix|length:]|lower == dupe.name|lower -%}
```

Comment:
This nested loop thing is O(n^2) right? I'm still learning Jinja, but seems it also supports dictionaries https://documentation.bloomreach.com/engagement/docs/datastructures#dictionaries

Reply:
@greg-finley correct it does result in n^2 and jinja does support dictionaries. How are you proposing we leverage a dictionary to replace the nested loops?

It may be worthwhile to move forward with this solution in the immediate so users can resolve the error and consider optimizing the macro down the road.

Reply:
Yep, 100%, let's ship it.

If we had the list of columns in a dictionary or set, we could O(1) look up whether the duplicate name exists vs looking through the column list again.